cloudinit/sources/DataSourceVMware.py

```diff
@@ -821,38 +821,73 @@ def is_valid_ip_addr(val):
     )
 
 
-def convert_to_netifaces_format(addr):
+def convert_to_netifaces_ipv4_format(addr):
     """
     Takes a cloudinit.netinfo formatted address and converts to netifaces
     format, since this module was originally written with netifaces as the
     network introspection module.
-    netifaces format:
+    netifaces ipv4 format:
     {
       "broadcast": "10.15.255.255",
       "netmask": "255.240.0.0",
       "addr": "10.0.1.4"
     }
-
-    cloudinit.netinfo format:
+    cloudinit.netinfo ipv4 format:
     {
       "ip": "10.0.1.4",
       "mask": "255.240.0.0",
       "bcast": "10.15.255.255",
       "scope": "global",
     }
     """
-    return {
-        "broadcast": addr["bcast"],
-        "netmask": addr["mask"],
-        "addr": addr["ip"],
+    addr_in_netifaces_format = {}
+    if "bcast" in addr:
+        addr_in_netifaces_format["broadcast"] = addr["bcast"]
+    if "mask" in addr:
+        addr_in_netifaces_format["netmask"] = addr["mask"]
+    if "ip" in addr:
+        addr_in_netifaces_format["addr"] = addr["ip"]
+    return addr_in_netifaces_format
+
+
+def convert_to_netifaces_ipv6_format(addr):
+    """
+    Takes a cloudinit.netinfo formatted address and converts to netifaces
+    format, since this module was originally written with netifaces as the
+    network introspection module.
+    netifaces ipv6 format:
+    {
+      "netmask": "ffff:ffff:ffff:ffff::/64",
+      "addr": "2001:db8:abcd:1234::1"
+    }
+    cloudinit.netinfo ipv6 format:
+    {
+      "ip": "2001:db8:abcd:1234::1/64",
+      "scope6": "global",
     }
+    """
+    addr_in_netifaces_format = {}
+    if "ip" in addr:
+        ipv6_with_netmask = addr["ip"]
+        ipv6_parts = ipv6_with_netmask.split("/")
+        ipv6_address = ipv6_parts[0]
```

Comment:
Is there any chance this will ever process anything without a / in it?  Would it make sense put these dereferences in a try/except group?

Reply:
Good catch.

It looks like when an interface is [point-to-point this is possible](https://github.com/canonical/cloud-init/blob/9c77b3896c903cadd966c71b7301593c06145520/cloudinit/netinfo.py#L89).

Reply:
Thanks for the review!
I've updated this function as Brett's suggestion as below.